syntax = "proto3";
package NUgc.NSecurity;

// The bundle contains enough information to generate a RecordIdentifier.
//
message TRecordIdentifierBundle {
    // A top-level container to which this record belongs. This prevents collisions across
    // applications.
    // Required, minimum 8 characters.
    string Namespace = 1;

    // The id of the user creating the record. Different aliases will produce different identifiers.
    // If no user id is available, some other agent id may be used, but then you must use a server
    // token to prevent malicious collisions.
    // Required, minimum 8 characters.
    string UserId = 2;

    // The id of the situation in which the record is generated, such as a request id.
    string ContextId = 3;

    // The id of the object that this record applies to, if any.
    string ParentId = 4;

    // Client-supplied data that can be used to distinguish this record from another one in
    // identical circumstances (e.g., two messages in a row). Client-generated random numbers,
    // counters, timestamps or even record contents may be used. Make sure retries do not produce
    // different values, though.
    string ClientEntropy = 5;
}
